src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/config/GerritJcascConfigurator.java

```diff
@@ -0,0 +1,295 @@
+        ) throws ConfiguratorException {
+            Mapping mapping;
+            if (config != null) {
+                mapping = config.asMapping();
+            } else {
+                mapping = Mapping.EMPTY;
+            }
+
+            try {
+                Sequence days = mapping.get(DAYS_OF_WEEK).asSequence();
+                int[] dayNumbers = new int[days.size()];
+                for (int i = 0; i < days.size(); i++) {
+                    dayNumbers[i] = nameToOrdinal(days.get(i).asScalar().getValue());
+                }
+
+                ArrayList<WatchTimeExceptionData.TimeSpan> timeSpans = new ArrayList<>();
+                for (CNode timeSpanNode : mapping.get(TIMES_OF_DAY).asSequence()) {
+                    Mapping tsm = timeSpanNode.asMapping();
+                    String from = tsm.get("from").asScalar().getValue();
+                    String to = tsm.get("to").asScalar().getValue();
+                    WatchTimeExceptionData.TimeSpan timeSpan = new WatchTimeExceptionData.TimeSpan(
+                            WatchTimeExceptionData.Time.createTimeFromString(from),
+                            WatchTimeExceptionData.Time.createTimeFromString(to)
+                    );
+                    timeSpans.add(timeSpan);
+                }
+                return new WatchTimeExceptionData(dayNumbers, timeSpans);
+            } catch (IllegalArgumentException ex) {
+                throw new ConfiguratorException(
+                        this,
+                        "Failed configuring " + getTarget() + " from " + mapping,
+                        ex
+                );
+            }
+        }
+
+        @Override
+        public WatchTimeExceptionData check(CNode config, ConfigurationContext context) throws ConfiguratorException {
+            // Just do what #configure does as long as #configure is a pure factory method
+            return configure(config, context);
+        }
+    }
+
+    /**
+     * Configure {@link com.sonymobile.tools.gerrit.gerritevents.watchdog.WatchTimeExceptionData.TimeSpan}.
+     */
+    @Extension
+    public static final class TimeSpanConfigurator implements Configurator<WatchTimeExceptionData.TimeSpan> {
+
+        @Override
+        public Class<WatchTimeExceptionData.TimeSpan> getTarget() {
+            return WatchTimeExceptionData.TimeSpan.class;
+        }
+
+        @Override
+        public Set<Attribute<WatchTimeExceptionData.TimeSpan, ?>> describe() {
+            return ImmutableSet.of(
+                    new Attribute<WatchTimeExceptionData.TimeSpan, String>("from", String.class).getter(
+                            ts -> ts.getFrom().getHourAsString() + ":" + ts.getFrom().getMinuteAsString()
+                    ),
+                    new Attribute<WatchTimeExceptionData.TimeSpan, String>("to", String.class).getter(
+                            ts -> ts.getTo().getHourAsString() + ":" + ts.getTo().getMinuteAsString()
+                    )
```

Comment:
Have you checked the correct indenting here?

Reply:
The codebase does not seem to be consistent on indenting following lines of a single expression broken over multiple lines. I do prefer indenting such lines by a multiple of normal indent (`4` vs `8`) in this case. `PluginConfig#setValues` alone mixes both stiles. :shrugs: